src/components/AuthInput.tsx

```diff
@@ -1,53 +1,61 @@
-import { useState } from "react";
+import { useState, type FocusEvent, type FocusEventHandler, type InputHTMLAttributes } from "react";
 import Input from "./Input";
 import icEyeVisible from "../assets/ic_eye_visible.svg";
 import icEyeInvisible from "../assets/ic_eye_invisible.svg";
 import "./AuthInput.css";
 
-function PasswordInput ({ name, className, onFocusout, valid, wrongMessage, ...props }) {
+interface PasswordProps extends InputHTMLAttributes<HTMLInputElement> {
+    name: string;
+    label: string;
+    className?: string; 
+    onFocusout?: FocusEventHandler; 
+    valid?: string; 
+    wrongMessage: string;
+}
+
+function PasswordInput ({ name, label, className='', onFocusout, valid, wrongMessage, ...props }: PasswordProps) {
     const [isVisible, setIsVisible] = useState(false);
     const onClick = () => {
         setIsVisible((prev) => !prev);
     }
     const passwordMatch = () => {
         try {
-            const password = document.querySelector("input#password");
-            const passwordCheck = document.querySelector("input#password-check");
+            const password = document.querySelector("input#password") as HTMLInputElement;
+            const passwordCheck = document.querySelector("input#password-check") as HTMLInputElement;
```

Comment:
💊 제안
타입 단언문은 타입 안정성에 좋지 않기 때문에 꼭 필요한 경우에만 사용하시는 것이 좋습니다. querySelector의 경우 해당하는 조건의 태그를 찾지 못하면 null을 반환할 수 있으므로 더욱 위험합니다~
이런 경우 아래처럼 작성하시는 것을 추천드립니다.

```suggestion
            // querySelector에 제네릭 타입에 HTMLInputElement임을 명시
            const password = document.querySelector<HTMLInputElement>("input#password"); 
            const passwordCheck = document.querySelector<HTMLInputElement>("input#password-check");

            if (!password || !passwordCheck) return; // null check
```